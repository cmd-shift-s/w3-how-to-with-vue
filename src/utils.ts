import MarkdownIt from 'markdown-it'
import hljs from 'highlight.js'
import 'highlight.js/styles/vs.css'

export const md = new MarkdownIt({
  highlight (code, lang) {
    return hljs.highlight(lang, code).value
  }
})
