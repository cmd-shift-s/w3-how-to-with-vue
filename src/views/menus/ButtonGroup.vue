<template lang="html">
  <page
    title="Button Group"
    :intro="intro"
    class="button-group-page">

  <template slot="header">
  <h2 class="title">Horizontal Button Groups</h2>
  <p>Group a series of buttons together on a single line in a button group:</p>
  </template>

  <sample>
    <button-group>
      <button v-for="name of buttons" v-text="name" :key="name"></button>
    </button-group>
  </sample>
  <br>
  <sample>
    <button-group is-fullwidth>
      <button v-for="name of buttons" v-text="name" :key="name"></button>
    </button-group>
  </sample>

  <template slot="examples">
  <hr>
  <h2 class="title">How To Create a Button Group</h2>
  <step no="1" html></step>
  <example-code :code="examples[0]"></example-code>
  <step no="2" css></step>
  <example-code :code="examples[1]"></example-code>
  <hr>
  <h2 class="title">Justified / Full-width Button Group:</h2>
  <example-code :code="examples[2]"></example-code>
  </template>

  </page>
</template>

<script>
import ButtonGroup from '@/components/menus/ButtonGroup.vue'

export default {
  name: 'button-group-page',
  data() {
    return {
      intro: 'a "button group" with CSS',
      buttons: ['Apple', 'Samsung', 'Sony'],
      examples: [
        [
          '``` html',
          '<div class="btn-group">',
          '  <button>Apple</button>',
          '  <button>Samsung</button>',
          '  <button>Sony</button>',
          '</div>',
          '```'
        ],
        [
          '``` css',
          '.btn-group button {',
          '  background-color: #4CAF50; /* Green background */',
          '  border: 1px solid green; /* Green border */',
          '  color: white; /* White text */',
          '  padding: 10px 24px; /* Some padding */',
          '  cursor: pointer; /* Pointer/hand icon */',
          '  float: left; /* Float the buttons side by side */',
          '}',
          '',
          '.btn-group button:not(:last-child) {',
          '  border-right: none; /* Prevent double borders */',
          '}',
          '',
          '/* Clear floats (clearfix hack) */',
          '.btn-group:after {',
          '  content: "";',
          '  clear: both;',
          '  display: table;',
          '}',
          '',
          '/* Add a background color on hover */',
          '.btn-group button:hover {',
          '  background-color: #3e8e41;',
          '}',
          '```'
        ],
        [
          '``` html',
          '<!-- Three buttons in a group -->',
          '<div class="btn-group" style="width:100%">',
          '  <button style="width:33.3%">Apple</button>',
          '  <button style="width:33.3%">Samsung</button>',
          '  <button style="width:33.3%">Sony</button>',
          '</div>',
          '',
          '<!-- Four buttons in a group -->',
          '<div class="btn-group" style="width:100%">',
          '  <button style="width:25%">Apple</button>',
          '  <button style="width:25%">Samsung</button>',
          '  <button style="width:25%">Sony</button>',
          '  <button style="width:25%">HTC</button>',
          '</div>',
          '```'
        ]
      ]
    }
  },
  components: {
    ButtonGroup
  }
}
</script>

<style lang="scss">
</style>
