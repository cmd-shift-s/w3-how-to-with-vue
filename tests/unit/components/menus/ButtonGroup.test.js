import { mount, renderer } from '@/tests/unit/utils'
import ButtonGroup from '@/components/menus/ButtonGroup.vue'

describe('ButtonGroup.vue', () => {
  it('render default', () => {
    const wrap = mount(ButtonGroup)

    expect(wrap.classes()).toContain('button-group')
    expect(wrap.isEmpty()).toBe(true)

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#is-fullwidth', () => {
    const wrap = mount(ButtonGroup, {
      context: {
        props: { isFullwidth: true }
      }
    })

    expect(wrap.classes()).toContain('is-fullwidth')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#is-vertical', () => {
    const wrap = mount(ButtonGroup, {
      context: {
        props: { isVertical: true }
      }
    })

    expect(wrap.classes()).toContain('is-vertical')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
