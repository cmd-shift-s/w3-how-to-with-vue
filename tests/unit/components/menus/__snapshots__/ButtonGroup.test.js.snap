// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup.vue props#is-fullwidth 1`] = `<div class="button-group is-fullwidth"></div>`;

exports[`ButtonGroup.vue props#is-vertical 1`] = `<div class="button-group is-vertical"></div>`;

exports[`ButtonGroup.vue render default 1`] = `<div class="button-group"></div>`;
